static/js/common.js

```diff
@@ -110,6 +110,24 @@ function toggle_password_visibility(password_field_id, password_selector, tippy_
     set_password_toggle_label(password_selector, label, tippy_tooltips);
 }
 
+// Function to hide the visibility toggle for MS Browsers
+function check_if_show_password_to_hide() {
+    const user_agent = window.navigator.userAgent.toLowerCase();
+
+    // Internet Explorer
+    const is_IE = false || user_agent.includes("trident");
+
+    // MS Edge
+    const is_edge = false || user_agent.includes("edge");
+
+    // Edge (based on chromium)
+    const is_edge_chromium = false || user_agent.includes("edg");
+
+    if (is_IE || is_edge || is_edge_chromium) {
+        $(".password_visibility_toggle").css("display", "none");
+    }
+}
```

Comment:
We need to find a better way to do these checks that is less hacky; this has a lot of risk of being wrong in corner cases.  I'm sure stackoverflow or a JS library can offer a correct test.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent

In the official documentation here, it is also mentioned to use this way. Let me find more ways and resource to this.
Also, is there any corner case in your mind right now?